nomad/structs/services.go

```diff
@@ -1562,21 +1502,8 @@ type ConsulExposePath struct {
 	ListenerPort  string
 }
 
-func exposePathsEqual(pathsA, pathsB []ConsulExposePath) bool {
-	if len(pathsA) != len(pathsB) {
-		return false
-	}
-
-LOOP: // order does not matter
-	for _, pathA := range pathsA {
-		for _, pathB := range pathsB {
-			if pathA == pathB {
-				continue LOOP
-			}
-		}
-		return false
-	}
-	return true
+func exposePathsEqual(a, b []ConsulExposePath) bool {
+	return set.From(a).Equal(set.From(b))
```

Comment:
This copies both `a` and `b`. Not a big deal in this case, but we may not want to make this our default pattern. A generic verison of the loop below called `SliceSetEq` or `ElementsEq` or something might be a better pattern to use in places like this where we don't have an existing `go-set.Set` type to work with. I would imagine our "set" slices usually have so few elements that even the pathological case of iterating over all of `b` for every element of `a` is faster than creating a new map.